designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,96 @@
+# Search CLI Tool for NuGet.exe and Dotnet.exe
+
+* Status: **Implementing**
+* Author(s): [Advay Tandon](https://github.com/advay26)
+* Issue: [9704](https://github.com/NuGet/Home/issues/9704) - Search CLI Tool for NuGet.exe and Dotnet.exe
+
+## Problem Background
+
+Search functionality from the command line for NuGet.exe and the .NET Core CLI has been a feature request from the NuGet open source community for some time. Additionally, the NuGet API is already equipped to handle search requests, so an extension of this service to command line users would be both practical to implement and also highly valuable.
+
+## Who are the customers
+
+This targets all NuGet.exe and .NET users that want to be able to search for packages from the command line, rather than using NuGet Gallery or VS PMUI. Also, while `nuget.exe list` allows users to search different package sources for packages with a search query, the results are returned sorted by the package Id, rather than by relevance, as proposed for this Search command. Further, the Search command would provide more information on the packages than List currently does.
+
+## Goals
+
+* Providing search functionality for NuGet packages through the commandline for NuGet.exe
+* Implementing a prototype of this feature for dotnet.exe
+* Displaying the search results to users in a way that maximizes the visibility of the most relevant results
+
+## Solution
+
+The search feature leverages the search service provided by the NuGet API. Search targets multiple sources and displays all of their results one after the other. The command line interface is simple: __./NuGet.exe search \<query\>__.
+
+```
+PS C:\> ./NuGet.exe search logging
+```
+
+Each search result displays the name, version, number of downloads, and a preview of the description of the package.
+
+```
+====================
+Source: NuGet.org
+--------------------
+> Microsoft.Extensions.Logging.Abstractions | 5.0.0-preview.6.20305.6 | Downloads: 345,145,935
```

Comment:
I really like this lines format. It lends itself to grep'ing very well since there is a well-known character `>` at the beginning of the line. Consider indenting the description lines by at least one space so that a description containing `>` is not a false positive. Also, the indenting might look nice a la `dotnet list package`

Reply:
I've included this change, but the PrintJustified method I used in my prototype needs some adapting to fit my needs here so I'm going to continue working on this.